src/cirrus/builtins/feeders/feed-rerun/lambda_function.py

```diff
@@ -18,22 +21,25 @@
 # Cirrus state DB
 statedb = StateDB()
 
-# logging
-logger = logging.getLogger("feeder.rerun")
+
+def publish_batch(messages):
+    params = {
+        "TopicArn": SNS_TOPIC,
+        "PublishBatchRequestEntries": [
+            {"Id": str(idx), "Message": msg} for idx, msg in enumerate(messages)
+        ],
+    }
+    return SNS_CLIENT.publish_batch(**params)
 
 
 def submit(payload_ids):
-    with batch_handler(
-        SNS_CLIENT.publish_batch,
-        {"TopicArn": SNS_TOPIC},
-        "PublishBatchRequestEntries",
-    ) as handler:
+    with batch_handler(publish_batch, {}, "messages", batch_size=10) as handler:
         for payload_id in payload_ids:
-            handler.add({"url": StateDB.get_payload_url(payload_id)})
+            handler.add(json.dumps({"url": StateDB.get_payload_url(payload_id)}))
```

Comment:
In running this I had an error because I wasn't constructing the entries properly (they needed to be dicts with `Id` and `Message` keys. I created the `publish_batch` function to handle building those entries.

For background, the `batch_handler` is a context manager factory function that returns a `BatchHandler` class (https://github.com/cirrus-geo/cirrus-geo/blob/main/src/cirrus/lib2/utils.py#L252), which runs the provided function for every `batch_size` items added to the handler. In the case that the batch size doesn't divide evenly into the number of items, the context manager will ensure any remaining items added but not processed will be flushed to the specified function at context exit.

I'm also using this in the `process` lambda to handle the batching of SQS message deletes in the case of partial success.